src/test/java/com/coveo/pushapiclient/ApiUrlTest.java

```diff
@@ -0,0 +1,94 @@
+package com.coveo.pushapiclient;
+
+import static org.junit.Assert.assertEquals;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+
+import org.junit.Before;
+import org.junit.Test;
+
+public class ApiUrlTest {
+
+    private ApiUrl defaultUrl;
+    private ApiUrl regionOnlyUrl;
+    private ApiUrl regionOnlyUrlCaseInsensitive;
+    private ApiUrl environmentOnlyUrl;
+    private ApiUrl environmentAndRegionUrl;
+    private ApiUrl streamURL;
+
+    @Before
+    public void setUp() throws MalformedURLException {
+        defaultUrl = new ApiUrl(
+                new URL("https://api.cloud.coveo.com/push/v1/organizations/my-org-id/sources/my-source-id/documents"));
```

Comment:
Should `ApiUrl` accept a string, and call `new URL` under the hood?

Reply:
Not sure about this one. 
The URL object is meaningful in its own and other classes (`CatalogSource`, `PushSource`) may depend on it as well:
```java
public CatalogSource(String apiKey, URL sourceUrl) {
 ApiUrl url = new ApiUrl(sourceUrl);
 ... 
```
We could convert every URL arguments with String, but I feel like it would only add unnecessary MalformUrlExceptions exceptions that the SDK would need to handle.